Interfaces/API/Dirac.py

```diff
@@ -993,6 +993,46 @@ def __printOutput(fd=None, message=''):
     else:
       print(message)
 
+  #############################################################################
+  def listCatalogDir(self, directoryLFN, printOutput=False):
+    """lists the contents of a directory in the DFC
+    example usage: dirac.listCatalogDir("/lz/data/test", printOutput=True)
+    param directoryLFN: LFN of the directory to be listed
+    type directoryLFN: string or list in LFN format
+    param printOutput: prints output in a more human readable form
+    type printOutput: Boolean
+    """
+    ret = self._checkFileArgument(directoryLFN, 'LFN')
+    if not ret['OK']:
+      return ret
+    fc = FileCatalog()
+    res = fc.listDirectory(directoryLFN)
+    if not res['OK']:
+      self.log.warn(res['Message'])
+      return res
+    if not res['Value']['Successful']:
+      self.log.warn("listCatalogDir failed for all LFNs (%s)." % directoryLFN)
+      return res
```

Comment:
Don't you want to return `S_ERROR` here?

Reply:
Initially I returned an error, but @sfayer pointed out to me that someone might want to use this function to test for the non-existence of a directory.